qbraid_qir/qasm3/visitor.py

```diff
@@ -1165,6 +1179,55 @@ def _visit_statement_block(block):
             one=lambda: _visit_statement_block(if_block),
         )
 
+    def _visit_forin_loop(self, statement: ForInLoop) -> None:
+        def _declare_loop_variable(statement: ForInLoop, init_exp: Expression):
+            self._visit_classical_declaration(
+                ClassicalDeclaration(statement.type, statement.identifier, init_exp)
```

Comment:
Just confirming, the type of `ForInLoop` would be the type of the iterator too? As in - 

```c
for int i in [0:2]{ 
    cx q[i], q[i+1];
 }
```
would mean that `ForInLoop.type` would be  `Qasm3IntType` correct?

Reply:
Yes, that's right :)